libs/FirmwareKit/tests/FirmwareKit_test.cpp

```diff
@@ -26,9 +27,11 @@ class FirmwareKitTest : public ::testing::Test
 	// void SetUp() override {}
 	// void TearDown() override {}
 
-	Version current_version = Version {1, 2, 0};
-
-	std::string current_version_str = OS_VERSION;
+	Version current_version = Version {
+		semver::version {OS_VERSION}.major,
+		semver::version {OS_VERSION}.minor,
+		semver::version {OS_VERSION}.patch,
+	};
```

Comment:
This is in conflict with issue

- https://github.com/leka/LekaOS/issues/1071

Reply:
yes and no -- the idea of removing semver was to remove complicated code just to get the version number.

semver is using `constexpr` almost everywhere, which means that the actual work is done at compile time and not run time, so the final binary should only have the actual major, minor and patch values.

this must be further investigated to be confirmed.

on the other hand, removing semver will mean finding another way to parse the OS_VERSION, either by using lines for each values or different files. A bit more complicated to handle cleanly in cmake.

also, semver is already being used in embedded code, so using it in unit tests is normal and cleans up things while we investigate further and decide to keep it or remove it for something else.